ethereum/geth_client.go

```diff
@@ -107,9 +104,7 @@ func NewGethClient(config Config) (Client, error) {
 		client:    ethclient.NewClient(c),
 		rpcClient: c,
 		host:      u,
-		accounts:  make(map[string]*bind.TransactOpts),
```

Comment:
Why are we removing this logic? It saves ~15secs on every tx prep.

Reply:
1. We shouldn't have cached this since it contains all latest transaction level details including gasPrice
2. Waiting 15 sec is better than not getting the tx throught at all :) 
3. Cache creates race conditions even with the lock since opts are being modified in multiple places now (the reason that made me look at this, but I thought why optimal gas price was static all the time when testing with mainnet before, this was the reason)

Reply:
I disagree on the removal of this. We can adapt the gasLimit but not the transactor per request. This cache is only per account and since the field that will be updated is the gasLimit then if we make a copy of the opts per request we do not have that potential problem.
If you want we can discuss this offline

Reply:
As agreed I will keep the cache while adding the other info ondemand on a copy.